### PR TITLE
api/client: close the returned io.ReadCloser

### PR DESCRIPTION
**Pull request by Mashimiao**
Signed-off-by: Ma Shimiao mashimiao.fnst@cn.fujitsu.com

### CODE REVIEW

**Comment by crosbymichael** [comment]
@Mashimiao nice find.  Thanks!

LGTM

File: api/client/commit.go [hunk 1]
```diff
@@ -67,6 +67,9 @@ func (cli *DockerCli) CmdCommit(args ...string) error {
 		}
 	}
 	stream, _, _, err := cli.call("POST", "/commit?"+v.Encode(), config, nil)
+	if stream != nil {
+		defer stream.Close()
+	}
```
Comment (runcom) [inline api/client/commit.go]: Can you add this `defer` after the err check so we'll be sure stream is not nil?

**Comment by calavera** [comment]
I agree with @runcom. In general, we should do the checks in order:

- First we check the error
- Second we defer the stream closing.

File: api/client/commit.go (hunk 1 above)
Comment (Mashimiao) [inline api/client/commit.go]: Thanks for your comment. Fixed

**Comment by Mashimiao** [comment]
@calavera Thanks for your comment. Fixed

**Comment by cpuguy83** [comment]
LGTM

**Comment by duglin** [comment]
LGTM just waiting for janky

**Comment by calavera** [comment]
awesome, thank you!

**Pull request merged**